turbo/adapter/ethapi/api.go

```diff
@@ -398,7 +399,9 @@ func newRPCTransaction(tx types.Transaction, blockHash common.Hash, blockNumber
 	switch t := tx.(type) {
 	case *types.LegacyTx:
 		chainId = types.DeriveChainId(&t.V)
-		result.ChainID = (*hexutil.Big)(chainId.ToBig())
+		if !chainId.Eq(u256.Num0) {
```

Comment:
- it has method `IsZero`
- please add comment like `// if a legacy transaction has an EIP-155 chain id, include it explicitly`
- 
- We have one more code for RLP parsing of txs: https://github.com/ledgerwatch/erigon-lib/blob/main/types/txn.go#L329 - do we need some changes there also?

Reply:
1) sw updated as suggested
2) comments added
3) In L329-L334 (is_protected case) the chainId is not read from transaction (because is not present) but retrieved in configuration